### PR TITLE
Change settings for make test to those for make main and add printing statistics after tests

### LINKED ISSUE #257: Common build settings for `make main` and `make test`
To improve compilation time we should get the following two done.

- [ ] Get `make test` to compile with the same settings as `make main`. This is done in _./makefile_
- [ ] Get *test/test.cpp* to output all statistics at the end.

### PR DESCRIPTION
**Pull request by AnnaBlume99**
Closes #257

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ TEST_C_FLAGS = "-g -O2"
 
 test:
 	@mkdir -p build/
-	@cd build/ && cmake -D CMAKE_BUILD_TYPE=Debug -D CMAKE_C_FLAGS=$(TEST_C_FLAGS) -D CMAKE_CXX_FLAGS=$(TEST_C_FLAGS) ..
+	@cd build/ && cmake -D CMAKE_BUILD_TYPE=Debug -D ADIAR_STATS_EXTRA=ON ..
```
Comment (SSoelvsten) [inline makefile:29]: Does the removal of the `TEST_C_FLAGS` suddenly make `make test` slower on your machine?

- yes: The variable should be renamed to `O2_FLAGS` and also used in target `main`.
- no: The dead variable `TEST_C_FLAGS` should also be removed

**Review by SSoelvsten — COMMENTED** [review]
There is a tiny question related to `TEST_C_FLAGS`.

Furthermore: one could argue these were two independent changes, and so should have had two independent commits. In general, if you ever need to write 'and' inside of the commit message, then you are committing too much at the same time.

File: makefile (hunk 1 above)
Comment (AnnaBlume99) [reply makefile:29]: Yes it does - I will fix it

**Review by SSoelvsten — APPROVED** [review]
Perfect. 🙂

**Pull request merged**